src/MLJTreeParzen.jl

```diff
@@ -198,40 +198,43 @@ end
 
 MLJTuning.default_n(tuning::MLJTreeParzenTuning, range) = 50
 
+vector(history::Nothing) = Trials.Trial[]
+vector(history::Vector) = history
+
+get_trialhist(history) =
+    map(history) do entry
+        trial_object = entry.metadata
+        measurement = entry.measurement[1]
+        # @ablaom asks "Is this deepcopy really necessary?":
+        completed_trial = deepcopy(trial_object) 
+        tell!(completed_trial, first(measurement))
+        completed_trial
+    end
 
-function MLJTuning.result(tuning::MLJTreeParzenTuning, history, state, evaluation, trial_in_progress)
-
-    completed_trial = deepcopy(trial_in_progress)
-    tell!(completed_trial, first(evaluation.measurement)) # `best` references the first element, which is a bit scary
-    # doing like this means we can keep the existing implementation for `best`
-    return (measure=evaluation.measure, measurement=evaluation.measurement, trial_object=completed_trial)
-
-end
-
-
-# this is the bit where we generate the internal history from the MLJ history
-# where we have injected the trial object ourselves using a modified `result` call
-# history is a vector of tuple of (model, result), where result is a namedtuple
-get_trialhist(history::Nothing) = Trials.Trial[]
-get_trialhist(history) = getindex.(getindex.(history, 2), :trial_object)
-
-histlen(history::Nothing) = 0
-histlen(history) = length(history)
-
-function MLJTuning.models!(
+function MLJTuning.models(
     strategy::MLJTreeParzenTuning,
     model,
     history,
-    state::MLJTreeParzenSpace,
+    state,
     remaining,
     verbosity,
-)::Vector{Tuple{Any, Trials.Trial}}
+)
+
+    space = state.space
+    trialhist = state.trialhist
+
+    num_hist = length(vector(history))
 
-    num_hist = histlen(history)
-    num_suggest = num_hist == 0 ? length(state.suggestions) : 0
+    # get an up-to-date the history of trial objects by appending to
+    # the trial object history stored in `state`:
+    Δhistory = view(vector(history), (length(trialhist) + 1):num_hist) # recent MLJ history
```

Comment:
I get (and appreciate) what you're doing here, but I'm not a fan of symbols in code -- please stick to alnums & underscores